webapp/components/expression/nodes/expressionNode.js

```diff
@@ -5,22 +5,33 @@ import * as R from 'ramda'
 import * as Expression from '@core/expressionParser/expression'
 import Binary from './binary'
 import Call from './call'
-import Group from './group'
+import Sequence from './sequence'
 import Identifier from './identifier'
 import Literal from './literal'
 import Logical from './logical'
 import Member from './member'
 
-const components = {
-  [Expression.types.Identifier]: Identifier,
-  [Expression.types.MemberExpression]: Member,
-  [Expression.types.Literal]: Literal,
-  // [Expression.types.ThisExpression]: thisExpression,
-  [Expression.types.CallExpression]: Call,
-  // [Expression.types.UnaryExpression]: unaryExpression,
-  [Expression.types.BinaryExpression]: Binary,
-  [Expression.types.LogicalExpression]: Logical,
-  [Expression.types.GroupExpression]: Group,
+const getComponent = (expressionNode) => {
+  const { type } = expressionNode
+  switch (type) {
+    case Expression.types.Identifier:
+      return Identifier
+    case Expression.types.MemberExpression:
+      return Member
+    case Expression.types.Literal:
+      return Literal
+    case Expression.types.CallExpression:
+      return Call
+    case Expression.types.BinaryExpression:
+      const { logical: logicalOperators } = Expression.operators
+      return [logicalOperators.or.value, logicalOperators.and.value].includes(expressionNode.operator)
+        ? Logical
+        : Binary
+    case Expression.types.SequenceExpression:
+      return Sequence
+    default:
+      throw new Error(`Expression type not supported: ${type}`)
+  }
 }
```

Comment:
I prefer the previous way because. is more Js style also it is easier to extend

components = {
}

getComponent = () => {
....
}

The part of BinaryExpression can check if the components return a component or a function evaluation this function.

Reply:
uhm, right I agree!